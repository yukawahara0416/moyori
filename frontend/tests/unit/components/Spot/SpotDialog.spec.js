import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Component from '@/components/Spot/SpotPostDialog.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

let wrapper
let getters
let actions
let vuetify
let store

beforeEach(() => {
  getters = {
    dialogSpotCreate: () => true
  }

  actions = {
    dialogOn: jest.fn(),
    dialogOff: jest.fn()
  }

  store = new Vuex.Store({
    getters,
    actions
  })

  vuetify = new Vuetify()

  wrapper = mount(Component, {
    localVue,
    store,
    vuetify,
    stubs: ['spot-form-container']
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('getters', () => {
  it('dialogSpotCreate', () => {
    expect(wrapper.vm.dialogSpotCreate).toEqual(getters.dialogSpotCreate())
  })
})

// describe('computed', () => {
//   it('dialog', () => {})
// })

describe('v-on', () => {
  const app = document.createElement('div')
  app.setAttribute('data-app', true)
  document.body.append(app)

  it('dialogOn', () => {
    const event = jest.fn()
    wrapper.setMethods({ dialogOn: event })
    wrapper.find('.v-dialog__container div').trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('actions', () => {
  it('dialogOn', () => {
    wrapper.vm.dialogOn()
    expect(actions.dialogOn).toHaveBeenCalled()
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
