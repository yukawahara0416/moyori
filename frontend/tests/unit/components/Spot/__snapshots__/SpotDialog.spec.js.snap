// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div
  class="v-dialog__container"
  role="dialog"
>
  <div />
</div>
`;
